string-constants-doc/string-constants/translation-guide.scrbl

```diff
@@ -0,0 +1,161 @@
+simplified) and Japanese. Dutch, Korean and Spanish have not been
+updated in a long while and they still need relicensing which makes
+them hidden in the interface.
+
+There are many languages missing plus some of the existing
+translations need some work for updates.
+
+If you are interested how much time and effort you will need to
+invest– currently it is roughly a 1400 string translation, of less
+than 6000 words — it is an easy-peasy deal.
+
+Here are the steps:
+
+@section{Preparation}
+
+Clone the
+@link["https://github.com/racket/string-constants"]{@tt{https://github.com/racket/string-constants}}
+repo. This is where you will work, translate and finally submit the
+translation via a pull request.
+
+Go to the
+@link["https://github.com/racket/string-constants/tree/master/string-constants-lib/string-constants/private"]{@tt{string-constants-lib/string-constants/private}}
+directory — it contains all the translations.  Start with copying
+the file
+@link["https://github.com/racket/string-constants/blob/master/string-constants-lib/string-constants/private/english-string-constants.rkt"]{@tt{english-string-constants.rkt}}
+to @tt{yourlanguage-string-constants.rkt}.
+
+Basically English is the master file containing the originals of the
+strings.  Therefore, start with a copy of that.  Note that if you
+continue to update and maintain your translation — English will be
+the file you will sync with.
+
+Once you have copied the file you will need to declare it so DrRacket
+knows it exists and has to be included with the rest of the
+translations in the official builds.  To do that just edit the file
+@link["https://github.com/racket/string-constants/blob/master/string-constants-lib/string-constants/string-constant.rkt"]{@tt{string-constants/string-constants-lib/string-constants/string-constant.rkt}}
+and add your language in two places:
+@racketblock[
+(require (prefix-in english: "private/english-string-constants.rkt")
+…
+          (prefix-in yourlanguage: "private/yourlanguage-string-constants.rkt"))
+]
+
+and
+@racketblock[
+(define built-in-string-constant-sets
+  (list
+   (make-sc 'english english:string-constants #f)
+…
+   (make-sc ' yourlanguage yourlanguage:string-constants #f)))
+]
+
+And with that you are done with the initial work — you have created
+a template for your language (still filled in with the English
+originals) and you have told Racket to use it. We are off to the races
+and all we need to do is fill in the translations.
+
+
+@; ----------------------------------------------------------------------
+
+@section{Translation}
+
+Open @tt{yourlanguage-string-constants.rkt} in your favorite text
+editor. It has a fairly trivial structure: it is basically a map — a
+long list of key-value pairs. The key is the identifier of the string
+— how Racket source code is going to refer to the message. Do not
+change these. Value is the translation. As you started with the
```

Comment:
Missing "The" in "The value is the translation."

Reply:
Done